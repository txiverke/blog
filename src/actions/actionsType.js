// @flow

export const LOAD_STATS_REQUEST = 'LOAD_STATS_REQUEST'
export const LOAD_STATS_SUCCESS = 'LOAD_STATS_SUCCESS'
export const LOAD_STATS_FAILURE = 'LOAD_STATS_FAILURE'

export const LOG_IN_USER_REQUEST = 'LOG_IN_USER_REQUEST'
export const LOG_IN_USER_SUCCESS = 'LOG_IN_USER_SUCCESS'
export const LOG_IN_USER_FAILURE = 'LOG_IN_USER_FAILURE'

export const IS_AUTHENTICATED_REQUEST = 'IS_AUTHENTICATE_REQUEST'
export const IS_AUTHENTICATED_SUCCESS = 'IS_AUTHENTICATE_SUCCESS'
export const IS_AUTHENTICATED_FAILURE = 'IS_AUTHENTICATE_FAILURE'

export const LOAD_USER_DATA_REQUEST = 'LOAD_USER_DATA_REQUEST'
export const LOAD_USER_DATA_SUCCESS = 'LOAD_USER_DATA_SUCCESS'
export const LOAD_USER_DATA_FAILURE = 'LOAD_USER_DATA_FAILURE'

export const UPDATE_USER_DATA_REQUEST = 'UPDATE_USER_DATA_REQUEST'
export const UPDATE_USER_DATA_SUCCESS = 'UPDATE_USER_DATA_SUCCESS'
export const UPDATE_USER_DATA_FAILURE = 'UPDATE_USER_DATA_FAILURE'

export const LOG_OUT_USER_REQUEST = 'LOG_OUT_USER_REQUEST'
export const LOG_OUT_USER_SUCCESS = 'LOG_OUT_USER_SUCCESS'
export const LOG_OUT_USER_FAILURE = 'LOG_OUT_USER_FAILURE'

export const LOAD_POST_DATA_REQUEST = 'LOAD_POST_DATA_REQUEST'
export const LOAD_POST_DATA_SUCCESS = 'LOAD_POST_DATA_SUCCESS'
export const LOAD_POST_DATA_FAILURE = 'LOAD_POST_DATA_FAILURE'

export const LOAD_POST_ITEM_REQUEST = 'LOAD_POST_ITEM_REQUEST'
export const LOAD_POST_ITEM_SUCCESS = 'LOAD_POST_ITEM_SUCCESS'
export const LOAD_POST_ITEM_FAILURE = 'LOAD_POST_ITEM_FAILURE'
export const UPDATE_POST_ITEM_SUCCESS = 'UPDATE_POST_ITEM_SUCCESS'

export const CREATE_POST_DATA_REQUEST = 'CREATE_POST_DATA_REQUEST'
export const CREATE_POST_DATA_SUCCESS = 'CREATE_POST_DATA_SUCCESS'
export const CREATE_POST_DATA_FAILURE = 'CREATE_POST_DATA_FAILURE'

export const UPDATE_POST_DATA_REQUEST = 'UPDATE_POST_DATA_REQUEST'
export const UPDATE_POST_DATA_SUCCESS = 'UPDATE_POST_DATA_SUCCESS'
export const UPDATE_POST_DATA_FAILURE = 'UPDATE_POST_DATA_FAILURE'

export const REMOVE_POST_DATA_REQUEST = 'REMOVE_POST_DATA_REQUEST'
export const REMOVE_POST_DATA_SUCCESS = 'REMOVE_POST_DATA_SUCCESS'
export const REMOVE_POST_DATA_FAILURE = 'REMOVE_POST_DATA_FAILURE'
