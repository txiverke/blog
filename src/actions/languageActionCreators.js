// @flow

import * as ACTION from './actionsType'

export const updateAppLanguage = (lang: string) => { 
  localStorage.setItem('xavierVilaTechLang', lang)
  return { type: ACTION.UPDATE_APP_LANGUAGE, payload: lang }
}
