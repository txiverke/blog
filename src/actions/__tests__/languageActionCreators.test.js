import * as action from '../languageActionCreators'
import * as type from '../actionsType'

describe('languageActionCreators', () => {
  it('should return UPDATE_APP_LANGUAGE action', () => {
    const lang = 'eng'
    expect(action.updateAppLanguage(lang)).toMatchSnapshot()
  })
})