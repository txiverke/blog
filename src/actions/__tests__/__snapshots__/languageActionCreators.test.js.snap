// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`languageActionCreators should return UPDATE_APP_LANGUAGE action 1`] = `
Object {
  "payload": "eng",
  "type": "UPDATE_APP_LANGUAGE",
}
`;
